lib/policies/userAgentPolicy.ts

```diff
@@ -61,7 +57,7 @@ export class UserAgentPolicy extends BaseRequestPolicy {
             request.headers = new HttpHeaders();
         }
 
-        if (!request.headers.get(this.headerKey)) {
+        if (!request.headers.get(this.headerKey) && this.headerValue && this.headerValue !== "") {
```

Comment:
Checking `this.headerValue` also ensures that `this.headerValue !== ""`. Please remove `&& this.headerValue !== ""`.
```suggestion
        if (!request.headers.get(this.headerKey) && this.headerValue) {
```